src/components/backend-ai-storage-proxy-list.ts

```diff
@@ -430,8 +430,9 @@ export default class BackendAIStorageProxyList extends BackendAIPage {
       // language=HTML
       html`
         <div id="controls" class="layout horizontal flex center" agent-id="${rowData.item.id}">
-          <mwc-icon-button class="fg blue controls-running" icon="assignment"
-                           @click="${(e) => this.showStorageProxyDetailDialog(rowData.item.id)}"></mwc-icon-button>
+          <mwc-icon-button class="fg blue controls-running" icon="assignment" 
+                          ?disabled=${Object.keys(JSON.parse(rowData.item.performance_metric)).length == 0 && JSON.parse(rowData.item.performance_metric).constructor == Object}
```

Comment:
If `JSON.parse(<string>)` is successful, its constructor must be an `Object`, otherwise, there will be `SyntaxError`. So, the second condition check after `&&` does not seem to be necessary. If `rowData.item.performance_metric` is not guaranteed to be a valid JSON string, you may write something like below:

```diff
    controlRenderer(root, column?, rowData?) {
+     let perfMetricDisabled;
+     try {
+       const perfMetric = JSON.parse(rowData.item.performance_metric);
+       perfMetricDisabled = Object.keys(perfMetric).length > 0 ? false : true;
+     } catch {
+       perfMetricDisabled = true;
+     }
      render(
```

Reply:
I haven't thought about the case that `JSON.parse(rowData.item.performance_metric)` causes an error. Thank you for letting me know what I missed. I change the code like you advised.